src/graphnet/utils/cluster/condor.py

```diff
@@ -0,0 +1,843 @@
+                self.logFileInfo.terminateTime
+                if self.logFileInfo.terminateTime != None
+                else "[Not parsed]"
+            )
+            returnString += "    Execution duration : %s\n" % (
+                self.logFileInfo.getExecuteDuration()
+                if self.logFileInfo.getExecuteDuration() != None
+                else "[Not parsed]"
+            )
+            returnString += "    Memory used        : %s\n" % (
+                "%i [MB]" % self.logFileInfo.memoryUsedMB
+                if self.logFileInfo.memoryUsedMB != None
+                else "[Not parsed]"
+            )
+            returnString += "    Memory requested   : %s\n" % (
+                "%i [MB]" % self.logFileInfo.memoryRequestedMB
+                if self.logFileInfo.memoryRequestedMB != None
+                else "[Not parsed]"
+            )
+            returnString += "    Memory allocated   : %s\n" % (
+                "%i [MB]" % self.logFileInfo.memoryAllocatedMB
+                if self.logFileInfo.memoryAllocatedMB != None
+                else "[Not parsed]"
+            )
+            returnString += "    Return code        : %s\n" % (
+                str(self.logFileInfo.returnCode)
+                if self.logFileInfo.returnCode != None
+                else "[Not parsed]"
+            )
+        else:
+            returnString += "  [Log file not parsed]\n"
+
+        if self.outFileInfo != None:
+            returnString += "  Out file info :\n"
+            returnString += "    Num sub jobs       : %s\n" % (
+                self.outFileInfo.numCommands
+                if self.outFileInfo.numCommands != None
+                else "[Not parsed]"
+            )
+        else:
+            returnString += "  [Out file not parsed]\n"
+
+        return returnString
+
+
+# Subclass to be filled with parsed out file
+class CondorJobMetricsOutFileInfo(object):
+    def __init__(self, outFilePath):
+        self.numCommands = None
+        self.parse(outFilePath)
+
+    def fullyParsed(self):
+        return self.numCommands != None
+
+    def parse(self, outFilePath):
+
+        # Check file exists, open it, and read into a string
+        if os.path.exists(outFilePath):
+            self.outFilePath = outFilePath
+            with open(self.outFilePath) as outFile:
+                outFileText = outFile.read()
+
+                # Parse num commands
+                self.numCommands = self.getNumCommandsFromFile(outFileText)
+
+    def getNumCommandsFromFile(self, outFileText):
+
+        # Split text into lines
+        lines = outFileText.split("\n")
+
+        # Find num commands line
+        key = "JOB INFO : Num commands = "
+        for line in lines:
+            keyPos = line.find(key)
+            if keyPos > -1:
+
+                # Num commands is rest of line after the key
+                return int(line[keyPos + len(key) :])
+
+        # If here, didn't manage to parse
+        return None
```

Comment:
I don't see this class referenced elsewhere (apart from one or more places that are not themselves references elsewhere)?
```suggestion
```